src/farcasterd/runtime.rs

```diff
@@ -151,35 +151,44 @@ impl Stats {
             Outcome::Abort => self.abort += 1,
         };
     }
+
     pub fn incr_initiated(&mut self) {
         self.initialized += 1;
     }
-    pub fn incr_awaiting_funding(&mut self, blockchain: &Blockchain) {
+
+    pub fn incr_awaiting_funding(&mut self, blockchain: &Blockchain, swapid: SwapId) {
         match blockchain {
-            Blockchain::Monero => self.awaiting_funding_xmr += 1,
-            Blockchain::Bitcoin => self.awaiting_funding_btc += 1,
-        }
+            Blockchain::Monero => self.awaiting_funding_xmr.insert(swapid),
+            Blockchain::Bitcoin => self.awaiting_funding_btc.insert(swapid),
```

Comment:
nit: since we've had issues with inconsistent incrementing/decrementing of funding info, I'd log warnings here if `swapid` is already present (likewise for removals).

Reply:
Done in [c6ab6df](https://github.com/farcaster-project/farcaster-node/pull/707/commits/c6ab6df1a2044e2a5a4056e677dfffd867683d92)